tests/unit/test_manof.py

```diff
@@ -0,0 +1,83 @@
+import mock
+
+from twisted.internet import defer
+from twisted.trial import unittest
+
+import manof.image
+import clients.logging
+
+logger = clients.logging.TestingClient('manof_unit').logger
+
+
+class ManofUnitTestCase(unittest.TestCase):
+
+    def __init__(self, *args, **kwargs):
+        super(ManofUnitTestCase, self).__init__(*args, **kwargs)
+
+    def setUp(self):
+        self._logger = logger
+        self._logger.info('Setting up unit test')
+
+    @property
+    def name(self):
+        return self.__class__.__name__
+
+    @defer.inlineCallbacks
+    def test_provision_pull(self):
+        self._logger.info('Testing manof provision with pull')
+        image = self._create_manof_image(
+            image_properties={
+                'image_name': 'test_image',
+            },
+            image_args={
+                'repository': None,
+                'tag_local': None,
+            })
+        yield self._assert_image_provision_commands(image, [
+            'docker pull docker.io/test_image'
+        ])
+
+    @defer.inlineCallbacks
+    def test_provision_build(self):
+        self._logger.info('Testing manof provision with build')
+        image = self._create_manof_image(
+            image_properties={
+                'image_name': 'test_image',
+                'context': 'test_image',
+            }
+        )
+        yield self._assert_image_provision_commands(image, [
+            'docker build --rm  --tag=test_image -f test_image/Dockerfile test_image'
+        ])
+
+    @defer.inlineCallbacks
+    def _assert_image_provision_commands(self, image, expected_commands):
+        self._logger.debug('Asserting image provision commands', image=image.name, commands=expected_commands)
+        self.patch(image, '_run_command', self._assert_expected_commands(expected_commands))
```

Comment:
that is one way to do it, another is to simply patch it with `mock.MagicMock` and assert it was called with. or even get the args it received.

Why im against this method? because every time I need to duplicate the exact command manof tried to execute. in unittest I let manof execute commands (as I mock them anyway) and what is important to me is more that it tried (and succeeded) to do it in the _right_ way. 

On the integration, I would check that the command achieved what I wanted (if it executed `docker build` I would check the image was built, rather than checking the command was `docker build x`)